api/v1alpha1/lvmcluster_types.go

```diff
@@ -35,6 +35,11 @@ type LVMClusterSpec struct {
 
 type DeviceClass struct {
 	// Name of the class, the VG and possibly the storageclass.
+	// Validations to confirm that this field can be used as metadata.name field in storageclass
+	// ref: https://github.com/kubernetes/apimachinery/blob/de7147/pkg/util/validation/validation.go#L209
+	// +kubebuilder:validation:MaxLength=245
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Pattern="^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$"
 	Name string `json:"name,omitempty"`
```

Comment:
I don't think this should have a "." in the name. The first part - "^[a-z0-9]([-a-z0-9]*[a-z0-9])?" -  is sufficient.

Reply:
It should follow this [pattern](https://github.com/kubernetes/apimachinery/blob/de7147de41c9331e4697399f48564f1d970cc01e/pkg/util/validation/validation.go#L178):
const dns1123LabelFmt string = "[a-z0-9]([-a-z0-9]*[a-z0-9])?"

Reply:
ack, I'll drop the subdomain format.